### PR TITLE
Add a catch-all "unknown" platform

### LINKED ISSUE #9: Add "other" platform
Currently this crate and any of its dependencies fail to compile on e.g. Wasm32 targets.

I believe it would be useful to add a catch-all implementation of `sys` that would just return "other" or "unknown" from all methods, to prevent at least compilation issues.

### PR DESCRIPTION
**Pull request by RReverser**
Fixes #9.

### CODE REVIEW

**Comment by Arcterus** [comment]
If we add this, I think it would be better to spit out a warning on compilation as well (so that someone will know if the library isn't going to work correctly on some platform).  You can maybe do this by adding an unused constant with the error message.  Might also want to make this a Cargo feature so that someone can make a compile fail if some platform isn't supported.

**Comment by RReverser** [comment]
> If we add this, I think it would be better to spit out a warning on compilation as well (so that someone will know if the library isn't going to work correctly on some platform).

For what it's worth, shimming like this is how most libraries are ported to wasm32 / wasi, and even how stdlib implements functions that can't be reasonably implemented on these targets otherwise, so I'm not sure that each of them emitting own warnings would be very useful...

Compiling to such platform is already a pretty explicit opt-in, and most users already expect system API support to vary across systems, as long as crates compile (and, when possible, don't panic in runtime too).

**Comment by RReverser** [comment]
Friendly ping?

**Comment by RReverser** [comment]
Ping...

**Review by sylvestre — APPROVED** [review]
Look good, thanks!

**Comment by sylvestre** [comment]
I will land it once the CI is green!

Will you need a new release on crates.io ?

**Comment by RReverser** [comment]
For now I'm only using it from a fork of `coreutils` with WASI support, but eventually I'd like to upstream my changes, so yeah, crates.io release would be useful.

Thanks!

**Comment by sylvestre** [comment]
First, I think I fixed the CI issue.

Then, do you think you could add a test like in?
https://github.com/uutils/platform-info/blob/ad1aa06ac9f300f802783a22b9c718518bba7a83/src/windows.rs#L378-L384
or it is too complex?

**Comment by RReverser** [comment]
Ah, sure, np. After porting all sorts of dependencies, adding a test is definitely not complex :)

**Comment by RReverser** [comment]
Done.

File: src/unknown.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    fn machine(&self) -> Cow<str> {
+        Cow::Borrowed("unknown")
+    }
+}
+
+#[test]
+fn test_unknown() {
+    let platform_info = PlatformInfo::new().unwrap();
+
+    assert_eq!(platform_info.sysname(), "unknown");
```
Comment (sylvestre) [inline src/unknown.rs:46]: this is a lot of unknown ;)

Comment (RReverser) [reply src/unknown.rs:46]: Still better than compilation errors :)

**Comment by sylvestre** [comment]
@RReverser done https://crates.io/crates/platform-info ;)

**Pull request merged**